lib/go-qmstr/master/common.go

```diff
@@ -173,21 +173,22 @@ func (gsp *genericServerPhase) GetSourceFileNodes(stream service.AnalysisService
 	offset, first := 0, 500
 
 	for {
-		filenodes, err := db.GetFileNodeLeaves(offset, first)
+		fileDatanodes, err := db.GetFileDataOfFileNodeLeaves(offset, first)
 		if err != nil {
 			return err
 		}
-		for _, filenode := range filenodes {
-
+		for _, fileDataNode := range fileDatanodes {
+			fileNodes, err := db.GetFileNodesByFileNode(&service.GetFileNodeMessage{FileNode: &service.FileNode{FileData: fileDataNode}}, 0, 500)
 			if gsp.server.currentPhase.GetPhaseID() == service.Phase_ANALYSIS {
-				filenode.Path = filepath.Join(gsp.masterConfig.Server.BuildPath, filenode.Path)
+				fileNodes[0].Path = filepath.Join(gsp.masterConfig.Server.BuildPath, fileNodes[0].Path)
```

Comment:
If an analyzer has analyzed file A with path /something/A it will not be able to store its results, when there is a filenode B with path /something/somethingnew/B, that is a copy of A and the query returns this file node B first.